test/xds/xds_server_rbac_test.go

```diff
@@ -660,6 +693,17 @@ func (s) TestRBACHTTPFilter(t *testing.T) {
 				}
 				// Toggle RBAC back on for next iterations.
 				envconfig.XDSRBAC = true
```

Comment:
I think it's worth considering adding some check to verify audit logging is not active till RBAC is toggled back.

Reply:
I think it ends up being somewhat implicit in the test - it makes 2 calls with RBAC on, then 2 calls with RBAC off, and the expected outcome for logging is `allowed: 2, denied: 0`

Reply:
Agree that this isn't the best as it requires holding some test state in your head as you look at it, but I hesitate to make large changes to this existing test